src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -852,19 +852,18 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
             }
             return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
         }
+        auto it_streams = options.find("GPU_THROUGHPUT_STREAMS");
         if (options.find("GPU_THROUGHPUT_STREAMS") != options.end()) {
-            try {
-                n_streams = options.find("GPU_THROUGHPUT_STREAMS")->second.as<uint32_t>();
-            } catch (...) {
-                try {
-                    std::string n_streams_str = options.find("GPU_THROUGHPUT_STREAMS")->second.as<std::string>();
-                    if (n_streams_str != CONFIG_VALUE(GPU_THROUGHPUT_AUTO)) {
-                        IE_THROW() << "[GPU_MAX_BATCH_SIZE] bad casting: GPU_THROUGHPUT_STREAMS should be either of uint32_t type or \"GPU_THROUGHPUT_AUTO\"";
-                    }
-                    n_streams = config.GetDefaultNStreamsForThroughputMode();
-                } catch (...) {
+            if (it_streams->second.is<uint32_t>()) {
+                n_streams = it_streams->second.as<uint32_t>();
+            } else if (it_streams->second.is<std::string>()) {
+                std::string n_streams_str = it_streams->second.as<std::string>();
+                if (n_streams_str != CONFIG_VALUE(GPU_THROUGHPUT_AUTO)) {
                     IE_THROW() << "[GPU_MAX_BATCH_SIZE] bad casting: GPU_THROUGHPUT_STREAMS should be either of uint32_t type or \"GPU_THROUGHPUT_AUTO\"";
                 }
+                n_streams = config.GetDefaultNStreamsForThroughputMode();
+            } else {
+                IE_THROW() << "[GPU_MAX_BATCH_SIZE] bad casting: GPU_THROUGHPUT_STREAMS should be either of uint32_t type or \"GPU_THROUGHPUT_AUTO\"";
             }
         }
```

Comment:
@vladimir-paramuzov please, have a look

Reply:
Looks good, but I think I can handle this part myself in my next PR (looks like this part requires additional modification related to API 2.0), if these plugin changes is not critical for this PR